src/m365/spo/commands/listitem/listitem-retentionlabel-ensure.ts

```diff
@@ -120,6 +125,13 @@ class SpoListItemRetentionLabelEnsureCommand extends SpoCommand {
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     try {
       const labelInformation = await this.getLabelInformation(args.options, logger);
+
+      if (args.options.assetId && !labelInformation.isEventBasedTag) {
+        throw `The label that's being applied is not an event-based label`;
```

Comment:
Could you remove this check? The AssetID field can just be updated, regardless whether the label in question is an event-based label.